src/index.js

```diff
@@ -144,10 +164,10 @@ const App = () => {
                     <ToDoItems todos={sortedTodos} onCompleteItem={handleCompleteItem} />
                 </tbody>  
             </table>
-            <input type='text' placeholder='event name' ref={toDoInput} />
-            <input type="date" min={tomrISO} ref={dueDateInput} /> <button onClick={handleAddItem}> Add </button>
+            <input type='text' placeholder='event name' ref={toDoInput} onChange={() => handleValidations()} />
+            <input type="date" min={tomrISO} ref={dueDateInput}  onChange={() => handleValidations()} /> <button onClick={handleAddItem} disabled={disableAddButton}> Add </button>
```

Comment:
If you need to use `onChange` then I would recommend just using `useState` to store the input value since you need to trigger re-render to check validation anyways. 

```js
const [todoInputValue, setTodoInputValue] = useState('');
...
const isAddButtonDisabled = !todoInputValue || !dateInputValue;
...
<input type='text' placeholder='event name' onChange={event => setTodoInputValue(e.target.value)} />
...
<button onClick={handleAddItem} disabled={isAddButtonDisabled}>Add</Button>

Reply:
So instead of setting the add button's useState through the inputs' onChange, just set a useState on the inputs to set/watch their states.
This is more direct since we are using useState anyways, why not just use it on the inputs.